irohad/torii/impl/command_service.cpp

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2017 Soramitsu Co., Ltd.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "torii/command_service.hpp"
+#include "common/types.hpp"
+
+namespace torii {
+
+  CommandService::CommandService(
+      iroha::model::converters::PbTransactionFactory &pb_factory,
+      iroha::torii::TransactionProcessor &txProccesor)
+      : pb_factory_(pb_factory), tx_processor_(txProccesor) {
+    // Notifier for all clients
+    tx_processor_.transaction_notifier().subscribe([this](auto iroha_response) {
+
+      // TODO: make for other responses
+      if (iroha:: instanceof
```

Comment:
Better solution to subsribe only for TransactionStatelessResponse, because this approach more satisfied to functional paradigm (operator composition), clear to understand and less dependent.
Easy way to do it make chain fiter -> map:
```C++
tx_processor_
      .transaction_notifier()
      .filter(/* check is object instance of StatelessResponse */)
      .map(/* convert to StatelessResponse*/)
      .subscribe(/* business logic here */)
```

Another way more complicated use flatmap, because it more effective combination of filter and map combined in one operator.